MouseUnSnag/GeometryUtil.cs

```diff
@@ -86,6 +93,22 @@ public static Point ClosestBoundaryPoint(this Rectangle r, Point p)
         /// <returns><see cref="Point"/></returns>
         public static Point OutsideDirection(Rectangle r, Point p) => Sign(OutsideDistance(r, p));
 
+        /// <summary>
+        /// In which direction(s) is(are) the rectangle r2 outside of the rectangle r1? If r2
+        /// overlaps r1, then this returns (0,0). Else X and/or Y can be either -1 or +1, depending
+        /// on which direction r2 is outside r1. For a rectangle to be diagonal to another rectangle,
+        /// it must not overlap horizontal or vertically.
+        /// </summary>
+        /// <param name="r"><see cref="Rectangle"/></param>
+        /// <param name="p"><see cref="Point"/></param>
+        /// <returns><see cref="Point"/></returns>
+        public static Point OutsideDirection(Rectangle r1, Rectangle r2) => (OverlapX(r1, r2), OverlapY(r1, r2)) switch {
```

Comment:
To account for different combinations of the `OverlapX` and `OverlapY` results, using a `switch` expression seems a bit convoluted. What about this?
```
var x = OverlapX(r1, r2) ? 0 : Math.Sign(r1.Left - r2.Left);
var y = OverlapY(r1, r2) ? 0 : Math.Sign(r1.Top - r2.Top));
return new Point(x, y);
```

Reply:
Consider calculating `r2 - r1` instead of `r1 - r2` for consistency with other direction-methods in this class.

Reply:
> Consider calculating `r2 - r1` instead of `r1 - r2`

Yes, `r1 - r2` is definitely backward.